app/components/ui/button.tsx

```diff
@@ -17,53 +17,99 @@ type BaseButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
   variant: (typeof variants)[keyof typeof variants];
   buttonStyle: (typeof buttonStyles)[keyof typeof buttonStyles];
   children: React.ReactNode;
+  icon?: true;
```

Comment:
Ha, I haven't seen this before! I believe you want `boolean` instead of `true` here? It probably works either way, but I haven't seen "only accepting true" before!

```suggestion
  icon?: boolean;
```